Handle R2 slowdown errors

xref https://github.com/earth-mover/arraylake/issues/5012

I don't think we are retrying on such errors. We should. This is from Arraylake integration testing: https://github.com/earth-mover/arraylake/actions/runs/22709034796/job/65842365445

```        
try:
>           result = await self._store.get(key, _byte_request_to_tuple(byte_range))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           icechunk.IcechunkError:   x error getting object from object store service error
E             | 
E             | context:
E             |    0: icechunk::storage::s3::get_object_range_read
E             |            with settings=Settings { concurrency: None, retries: Some(RetriesSettings { max_tries: Some(50), initial_backoff_ms: None, max_backoff_ms: None }), unsafe_use_conditional_update:
E             | None, unsafe_use_conditional_create: None, unsafe_use_metadata: None, storage_class: None, metadata_storage_class: None, chunks_storage_class: None, minimum_size_for_multipart_upload: None }
E             | key="bcdnzxq0_integration_test_repo_r2_hmac_muyqojxu_kynhuk_sync_quickstart/chunks/KEZRVSDN24GYPACV914G" range=0..371113
E             |              at icechunk/src/storage/s3.rs:1072
E             |    1: icechunk::storage::s3::fetch_chunk
E             |            with id=KEZRVSDN24GYPACV914G range=0..371113
E             |              at icechunk/src/storage/s3.rs:684
E             |    2: icechunk::asset_manager::fetch_chunk
E             |            with chunk_id=KEZRVSDN24GYPACV914G range=0..371113
E             |              at icechunk/src/asset_manager.rs:361
E             |    3: icechunk::store::get
E             |            with key="ersstv5/sst/c/2/0/0" byte_range=From(0)
E             |              at icechunk/src/store.rs:198
E             | 
E             |-> error getting object from object store service error
E             |-> service error
E             |-> unhandled error (ServiceUnavailable)
E             `-> Error { code: "ServiceUnavailable", message: "Reduce your rate of simultaneous reads on the same object." }
```

--- a/icechunk/src/storage/s3.rs
+++ b/icechunk/src/storage/s3.rs
@@ -247,7 +247,8 @@ pub async fn mk_client(
     // Note R2 sends 429 for "slowdown" while S3 sends 503.
     //   - R2: https://developers.cloudflare.com/r2/api/error-codes/
     //   - S3: https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html
-    static RETRY_CODES: &[u16] = &[408, 429];
+    // Tigris can occasionally respond with 499: "Client Closed Request"
+    static RETRY_CODES: &[u16] = &[408, 429, 499];
     // This confusingly named `retry_classifier` method ends up calling
     // `push_retry_classifier` after wrapping our custom classifier in `SharedRetryClassifier`.
     // Ultimately, this is a push on to a `Vec<SharedRetryClassifier>`, and is thus additive

